core/blockchain.go

```diff
@@ -1813,7 +1815,7 @@ func (bc *BlockChain) insertSideChain(block *types.Block, it *insertIterator) (i
 		// memory here.
 		if len(blocks) >= 2048 || memory > 64*1024*1024 {
 			log.Info("Importing heavy sidechain segment", "blocks", len(blocks), "start", blocks[0].NumberU64(), "end", block.NumberU64())
-			if _, err := bc.insertChain(blocks, false); err != nil {
+			if _, err := bc.insertChain(blocks, false, false); err != nil {
```

Comment:
You should also pass the `isCatalyst` mode into the `insertSideChain ` and use it here. 

Because it's also possible that CL gives us a block with pruned ancestor. In this case all the ancestors need to be inserted first, then the latest block given by CL. It's why I have a function called `recoverAncestor`.